### DISCUSSION TITLE
How to populate the 'management' IPv4 for a virtual machine

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#10100] [General]
So I'm not quite understanding how I'm supposed to populate the "Primary IPv4" drop down under 'Management' when creating a new Virtual Machine.
Even when I look at the demo site, it doesn't have any VM's with their IPv4 address populated.

I have created every pre-requisite I believe I need to.
a cluster group
a cluster type
a cluster
vlans, IP prefixes, and even created IP addresses.

What further step is needed in order to select an IP address for my virtual machines?

**Comment by ghost** [comment]
Figured it out. You have to first create an interface on the virtual machine, then assign that interface an IP. You can then elect to check the box to make it the primary IP of the virtual machine.

Also, if you haven't already created an IP address in the IPAM section for this IP address, doing the above steps will create it for you.

**Reply by candlerb** [reply]
That's it.  You can use a custom script to do this all in one go:
https://github.com/netbox-community/reports/blob/master/scripts/create_vm.py
